pyvista/utilities/transformations.py

```diff
@@ -0,0 +1,39 @@
+"""Transformations leveraging transforms3d library."""
+import numpy as np
+import transforms3d as tf3d
+
+
+def axis_angle_rotation(axis, angle, point=None, deg=True):
+    """Return a 4x4 matrix for rotation about an axis by given angle, optionally about a given point."""
+    if deg:
+        angle *= np.pi / 180
+    return tf3d.axangles.axangle2aff(axis, angle, point=point)
+
+
+def reflection(normal, point=None):
+    """Return a 4x4 matrix for reflection across a normal about a point."""
+    return tf3d.reflections.rfnorm2aff(normal, point=point)
+
+
+def apply_transformation_to_points(transformation, points, inplace=False):
+    """Apply a given transformation matrix (3x3 or 4x4) to a set of points."""
+    if transformation.shape not in ((3, 3), (4, 4)):
+        raise RuntimeError('`transformation` must be of shape (3, 3) or (4, 4)')
+
+    if transformation.shape[1] == 4:
+        # a stack is a copy
+        points_2 = np.hstack((points, np.ones((len(points), 1))))
```

Comment:
Did some minor benchmarking and it turns out using `np.empty` is slightly faster (tested on `numpy==1.20` and `numpy==1.19.5`.)

```py
"""
In [6]: timeit np.hstack((np.ones((n, 1)), points))
7.61 ms ± 20.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


In [9]: timeit alt_stack(points)
6.78 ms ± 23.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

"""

import numpy as np

n = 1000000
points = np.random.random((n, 3))

stack = np.hstack((np.ones((len(points), 1)), points))


def alt_stack(points):
    stack = np.empty((len(points), 4))
    stack[:, 0] = 1
    stack[:, 1:] = points
    return stack
```

```suggestion
        points_2 = np.empty((len(points), 4), dtype=points2.dtype)
        points_2[:, 0] = 1
        points_2[:, 1:] = points
```

Reply:
Thanks for the testing! I believe there is a typo here on the last line of your suggestion (last column should be assigned 1, not -1), so I won't commit as-is, but I will incorporate this.

Reply:
Oops!  Yes, should be 1.  Good catch.